### DISCUSSION TITLE
dev services `init-commands` produces warnings

### DISCUSSION THREAD

**Discussion by fredofm** [quarkiverse/quarkus-vault#93] [Q&A] [answered]
Hi all,

I'm trying to use "io.quarkiverse.vault:**quarkus-vault:**2.1.0" with dev services in dev mode (quarkus:dev).

The vault docker image is started automatically as expected and obviously the server has no secrets inside.

So I decided to use the property "quarkus.vault.devservices.init-commands: 'vault kv put secret/config password="p@$$w0rd"" to populate secrets at the startup.

And it seems that the command is not ok for quarkus (I tried the command in the started docker container and it works).

The problem is that the warning message with the command information is only shown for a millisecond and then it's collapsed / overwritten with the next console information.

I was looking the way to log that information to a file or configure the console to avoid overwriting the information but I didn't find any clue on how to do it.

### Questions;

- What is the right way to populate your vault when using dev services?.
- Is there any way to avoid overwriting the console information?

### Steps to reproduce:
I'm using VSCode terminal to execute it, but I also checked with the linux console (WSL) with the same result.

Quarkus version: 2.14.0.Final

**pom.xml** <--- include the vault dependency in your pom file
    <dependency>
      <groupId>io.quarkiverse.vault</groupId>
      <artifactId>quarkus-vault</artifactId>
      <version>2.1.0</version>
    </dependency>

**application.yml** <--- include init-command option in your yaml file
quarkus:
  vault:
    devservices:
      init-commands: "vault write secret/config password=\"p@$$w0rd\""

**output**
As you can see in the picture the Vault information (light green) is overwritten by other information and it's imposible to know what is happening. 

"**read     Read**" and "**Vault server    stat**" is part of the vault warning message that was overwritten.

![image](https://user-images.githubusercontent.com/18413150/202137088-461014cf-f77c-4e52-a827-c4637e3438f2.png)

```
Amazon Dev Services for localstack (dynamodb) starting: 2022-11-16 08:58:21,069 INFO  [🐳 [localstack/localstack:latest]] (build-8) Container localstack/localstack:latest is starting: 105599d2922a239d738b9839a127f01493c631d51ac3d7ee4820f39e3140b993
2022-11-16 08:58:32,433 INFO  [io.quarkus] (Quarkus Main Thread) bootloader 0.0.1-SNAPSHOT on JVM (powered by Quarkus 2.14.0.Final) 2022-11-16 08:58:32,437 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.s:    read        Read2022-11-16 08:58:32,440 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [amazon-dynamodb, amazon-dynamodb-enhanced, cdi, config-yaml, micrometer, resteasy, resteasy-jackson, smallrye-context-propagation, smallrye-openapi, swagger-ui, vault, vertx]Vault server    stat2022-11-16 09:50:58,138 INFO  [localstack] (docker-java-stream-761798948) [105599d2922a] Waiting for all LocalStack services to be ready2022-11-16 09:50:58,144 INFO  [localstack] (docker-java-stream-761798948) [105599d2922a] 2022-11-16 07:58:23,029 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to
```

**Comment by kdubb** [comment, marked as answer]
The command issue is that you should only provide the arguments to the vault program because it's already running; you need to drop the `vault`.

With regard to logging, this is how dev services logging is designed to work, I believe it collapses everything unless the service cannot be started; in this case the service is still started. I agree its not ideal.

**Reply by fredofm** [reply]
Thanks a lot for your tips. I'will check it.

**Reply by fredofm** [reply]
Hi again,

I'm checking the suggestion from @kdubb regarding the init-commands parameter with the following unsuccessful results:

Context:

- As suggested, after removing "vault" from the command the warning message disappears.
- I'm running quarkus with dev profile (mvn quarkus:dev)
- When starting the vault dev services two secret storages are created by default:

![image](https://user-images.githubusercontent.com/18413150/202472877-f02b5cf7-77e7-4184-ab22-288045c914b8.png)

- If I execute the command in the docker console itself, the new value is added with no issue:

```
/ # vault write secret/config password="pp"
Success! Data written to: secret/config
```

Tests:

based on the following application.yml:

```
quarkus:
  "%dev":
    vault:
      devservices:
        init-commands: 'write secret/config password="p@$$w0rd"'
```

I tried the following init-commands options:

- init-commands: 'write secret/config password="p@$$w0rd"'
- init-commands: "write secret/config password=\\"p@$$w0rd\\""
- init-commands: 'kv put secret/config password="p@$$w0rd"'
- init-commands: "kv put secret/config password=\\"p@$$w0rd\\""
- init-commands: 'write secret password="p@$$w0rd"'
- init-commands: "write secret password=\\"p@$$w0rd\\""
- init-commands: 'kv put secret password="p@$$w0rd"'
- init-commands: "kv put secret password=\\"p@$$w0rd\\""

But the secret storage is always empty.

Could you give me a clue about what I'm doing wrong?.

**Reply by fredofm** [reply]
OMG! .... my apologies.

I just realised that I have a mistake in the application.yml, "quarkus:" is no under "%dev" section ... -_-

I fixed it and now it works!!! , sorry for the inconveniences. :(
